server/netty-server/zio/src/main/scala/sttp/tapir/server/netty/zio/NettyZioServerOptions.scala

```diff
@@ -40,17 +39,18 @@ object NettyZioServerOptions {
       createOptions = (ci: CustomiseInterceptors[RIO[R, *], NettyZioServerOptions[R]]) => default(ci.interceptors)
     ).serverLog(defaultServerLog[R])
 
-  private val log: Logger = Logger[NettyZioServerInterpreter[Any]]
-
   def defaultServerLog[R]: DefaultServerLog[RIO[R, *]] = DefaultServerLog(
     doLogWhenReceived = debugLog(_, None),
     doLogWhenHandled = debugLog[R],
     doLogAllDecodeFailures = debugLog[R],
-    doLogExceptions = (msg: String, ex: Throwable) => ZIO.succeed { log.warn(msg, ex) },
+    doLogExceptions = (msg: String, ex: Throwable) => ZIO.logErrorCause(msg, Cause.die(ex)),
     noLog = ZIO.unit
   )
 
   private def debugLog[R](msg: String, exOpt: Option[Throwable]): RIO[R, Unit] =
-    ZIO.succeed(NettyDefaults.debugLog(log, msg, exOpt))
+    exOpt match {
```

Comment:
**suggestion:** Shouldn't this method checek `logger.isDebugEnabled()` as well?

Reply:
I don't think there's an obvious way to check the log level with ZIO, plus it wouldn't really make sense as everything is lazily evaluated, and we need to return `ZIO.unit` one way or another. The only unnecessary cost we pay here is the pattern matching on `exOpt`.

Reply:
Right, it's ZIO, I overlooked that 👍